src/dapp/libexec/dapp/dapp-build

```diff
@@ -31,12 +59,24 @@ else
   (set -x; dapp clean)
 fi
 
-find "${DAPP_SRC}" -name '*.sol' | while read -r x; do
-  dir=${x%\/*}
-  dir=${dir#$DAPP_SRC}
-  dir=${dir#/}
-  mkdir -p "$DAPP_OUT/$dir"
-  (set -x; solc "${opts[@]}" --abi --bin --bin-runtime --metadata /=/ -o "$DAPP_OUT/$dir" "$x")
-  json_file=$DAPP_OUT/$dir/${x##*/}.json
-  (set -x; solc "${opts[@]}" "${json_opts[@]}" /=/ "$x" >"$json_file")
-done
+mkdir -p "$DAPP_OUT"
+mapfile -t files < <(find "${DAPP_SRC}" -name '*.sol')
+(set -x; solc "${opts[@]}" "${json_opts[@]}" /=/ "${files[@]}" > "${DAPP_JSON}")
+
+if [ "$DAPP_BUILD_EXTRACT" ]; then
+  mapfile -t contracts < <(<"$DAPP_JSON" jq '.contracts|keys[]' -r | sort -u -t: -k2 | sort)
+  data=$(<"$DAPP_JSON" jq '.contracts' -r)
+  count=1
+  total=${#contracts[@]}
+  for path in "${contracts[@]}"; do
+    info -ne "Extracting build data... [$count/$total]\\r"
+    ((count++))
+    name="${path#*:}"
+    contract=$(echo "$data" | jq '.[''"'"$path"'"'']')
+    echo "$contract" | jq '.["abi"]' -r > "$DAPP_OUT/$name.abi"
+    echo "$contract" | jq '.["bin"]' -r > "$DAPP_OUT/$name.bin"
+    echo "$contract" | jq '.["bin-runtime"]' -r > "$DAPP_OUT/$name.bin-runtime"
+  done
+  echo
+  info "Used --extract, build data can be extracted from $DAPP_JSON for faster builds!\\n"
```

Comment:
I don't understand what this means: `--extract` extracts right? So what does it mean that one can further extract for faster builds?

Reply:
It means that it's faster to not use `--extract`. Rather, users would ideally get the specific things that they need from the .sol.json.

Reply:
I think this message doesn't make a lot of sense. Users have to opt-in to the slower behaviour, so will probably have executed without `--extract` before trying it, and will be able to see the speed difference by themselves.

Reply:
So just have no message?